Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -147,4 +217,113 @@ public struct FindOneAndUpdateOptions: Encodable {
 
     /// An optional `WriteConcern` to use for the command.
     public let writeConcern: WriteConcern?
+
+    fileprivate func asOpts() throws -> FindAndModifyOptions {
+        return try FindAndModifyOptions(arrayFilters: arrayFilters, bypassDocumentValidation: bypassDocumentValidation,
+                                        collation: collation, maxTimeMS: maxTimeMS, projection: projection,
+                                        returnDocument: returnDocument, sort: sort, upsert: upsert,
+                                        writeConcern: writeConcern)
+    }
+
+    /// Convenience initializer allowing any/all parameters to be omitted/optional
+    public init(arrayFilters: [Document]? = nil, bypassDocumentValidation: Bool? = nil, collation: Document? = nil,
+                maxTimeMS: Int64? = nil, projection: Document? = nil, returnDocument: ReturnDocument? = nil,
+                sort: Document? = nil, upsert: Bool? = nil, writeConcern: WriteConcern? = nil) {
+        self.arrayFilters = arrayFilters
+        self.bypassDocumentValidation = bypassDocumentValidation
+        self.collation = collation
+        self.maxTimeMS = maxTimeMS
+        self.projection = projection
+        self.returnDocument = returnDocument
+        self.sort = sort
+        self.upsert = upsert
+        self.writeConcern = writeConcern
+    }
+}
+
+/// A class wrapping a `mongoc_find_and_modify_opts_t`, for use with `MongoCollection.findAndModify`
+private class FindAndModifyOptions {
+    // an `OpaquePointer` to a `mongoc_find_and_modify_opts_t`
+    var _options: OpaquePointer?
+
+    init() {
+        self._options = mongoc_find_and_modify_opts_new()
+    }
+
+    // swiftlint:disable:next cyclomatic_complexity
+    init(arrayFilters: [Document]? = nil, bypassDocumentValidation: Bool? = nil, collation: Document?,
+         maxTimeMS: Int64?, projection: Document?, remove: Bool? = nil, returnDocument: ReturnDocument? = nil,
+         sort: Document?, upsert: Bool? = nil, writeConcern: WriteConcern?) throws {
+        self._options = mongoc_find_and_modify_opts_new()
+
+        if let bypass = bypassDocumentValidation,
```

Comment:
Would it make sense to keep the constant named the same as the argument? I recall doing `if let tagSets = tagSets` for ReadPreference.

If this was just done to be more concise, no worries.